examples/ai-agent-tool-kit/ai_agent_tool_kit.bal

```diff
@@ -0,0 +1,107 @@
+import ballerina/ai;
+import ballerina/io;
+import ballerina/time;
+import ballerina/uuid;
+
+type Task record {|
+    string description;
+    time:Date dueBy?;
+    time:Date createdAt = time:utcToCivil(time:utcNow());
+    time:Date completedAt?;
+    boolean completed = false;
+|};
+
+// A tool kit to manage a set of tasks.
+public isolated class TaskManagerToolkit {
+    *ai:BaseToolKit;
+    
+    private final map<Task> tasks = {};
+
+    // The `getTools` method describes the tools provided by this tool kit.
+    public isolated function getTools() returns ai:ToolConfig[] => [
+        // The JSON schema for the `addTask` tool.
+        {
+            name: "addTask",
+            description: "Add a new task to the to-do list",
+            parameters: {
+                'type: "object",
+                properties: {
+                    description: {'type: "string", description: "Description of the task"},
+                    dueBy: {
+                        'type: "object",
+                        description: "Due date for the task",
+                        properties: {
+                            year: {'type: "integer"},
+                            month: {'type: "integer"},
+                            day: {'type: "integer"}
+                        },
+                        required: ["year", "month", "day"]
+                    }
+                },
+                required: ["description"]
+            },
+            // Refer to the method that is this tool.
+            caller: self.addTask
+        },
+        {
+            name: "listTasks",
+            description: "List all current tasks",
+            caller: self.listTasks
+        }
+    ];
```

Comment:
```suggestion
    public isolated function getTools() returns ai:ToolConfig[] => ai:getToolConfigs([self. addTask, self. listTasks]);
```

While users can define the full schema themselves, the helper method suggested above offers a simpler way to create it.